lib/web_ui/lib/src/engine/text/paragraph.dart

```diff
@@ -431,9 +448,18 @@ class EngineParagraph implements ui.Paragraph {
 
   @override
   ui.TextRange getLineBoundary(ui.TextPosition position) {
-    // TODO(flutter_web): https://github.com/flutter/flutter/issues/39537
-    // Depends upon LineMetrics measurement.
-    return null;
+    final List<EngineLineMetrics> lines = _measurementResult.lines;
+    if (lines != null) {
+      final int offset = position.offset;
+
+      for (int i = 0; i < lines.length; i++) {
```

Comment:
Would binary search be overkill in this case?

Reply:
Yes :)

Most paragraphs are 1-4 lines.